packages/logger/src/formatter/LogFormatter.ts

```diff
@@ -104,16 +104,30 @@ abstract class LogFormatter {
    * @param error - Error to format
    */
   public formatError(error: Error): LogAttributes {
-    return {
-      name: error.name,
+    const { name, message, stack, cause, ...errorAttributes } = error;
+    const formattedError: LogAttributes = {
+      name,
       location: this.getCodeLocation(error.stack),
-      message: error.message,
-      stack: error.stack,
+      message,
+      stack,
       cause:
         error.cause instanceof Error
           ? this.formatError(error.cause)
           : error.cause,
     };
+    for (const key in error) {
+      if (
+        typeof key === 'string' &&
+        key !== 'name' &&
+        key !== 'message' &&
+        key !== 'stack' &&
+        key !== 'cause'
+      ) {
+        formattedError[key] = (errorAttributes as Record<string, unknown>)[key];
+      }
```

Comment:
Can we simplify it a bit? 
```suggestion
      if (typeof key === 'string' && !['name', 'message', 'stack', 'cause'].includes(key)) {
          formattedError[key] = (errorAttributes as Record<string, unknown>)[key];
      }
```